### PR TITLE
Add Hector TOR-2CRV APY

### PR DESCRIPTION
**Pull request by 0xCmdrKeen**
Adds Hector Curve TOR-2pool APY to the API.

### Overview

TOR is a stablecoin made by Hector DAO, [backed by HEC treasury reserves][1]. TOR can be [minted by burning HEC][2], and redeemed back for an equivalent value in HEC. TOR aims to achieve a 1:1 peg to the US Dollar.

Liquidity is provided via a [Curve pool][3], which lets users trade TOR against USDC and DAI. Liquidity provision for this pool is [incentivized by Hector][4], which pays rewards in WFTM in exchange for staking Curve LP tokens. 

[1]: https://docs.hector.finance/the-tor-stablecoin/the-tor-stablecoin
[2]: https://docs.hector.finance/the-tor-stablecoin/how-tor-works
[3]: https://ftm.curve.fi/factory/62
[4]: https://app.hector.finance/#/farming

### Contracts and Safety

- [TOR stablecoin](https://ftmscan.com/address/0x74E23dF9110Aa9eA0b6ff2fAEE01e740CA1c642e#code): mostly standard ERC20, with minting rights delegated to 
- [TOR Minter](https://ftmscan.com/address/0x9b0c6FfA7d0Ec29EAb516d3F2dC809eE43DD60ca#code): custom contract, owned by multisig (below) makes use of 
- [Mint/redeem strategy](https://ftmscan.com/address/0x45ac684b6b9ee1a8647f51170c90c8f943d002e3#code): owned by 
- [Multisig](https://ftmscan.com/address/0x814750c7e8b575b2b3da0d2f90eb689a1d665eb0#code): standard Gnosis multisig wallet which owns minter and mint strategy
- [Curve pool](https://ftmscan.com/address/0x24699312cb27c26cfc669459d670559e5e44ee60#code): standard Curve factory pool pairing TOR with the USDC+DAI metapool
- [Staking contract](https://ftmscan.com/address/0x61b71689684800f73ebb67378fc2e1527fbdc3b3#code): standard StakingRewards contract with variable emission rate

Contracts have been [audited by Certik](https://www.certik.com/projects/tor), additional comments from the developers can be found [here](https://medium.com/@HectorDAO/tor-has-been-audited-by-certik-88f0125c17ee).

The most glaring issue are the global parameters that can be set by the owner, including the replacement of the mint/redeem strategy. Certik recommended a combination of timelock and multisig to control those risks, but Hector decided to only implement multisig for now, saying it’s not a good fit for the project, because they need the flexibility to avert potential price manipulation attacks.

### SAFU Checklist

[x] Contracts have been verified in the blockexplorer
[x] Enough liquidity for swapping farm token rewards: yes (rewards paid in WFTM)
[x] Rug/migrator functions are either completely removed or timelocked sufficently: (none)
[x] Farm token emission rates have to be timelocked: no farm token pairs are being vaulted
[x] All proxy implementation changes must be timelocked: no proxies

Vault Strategy: https://github.com/beefyfinance/beefy-contracts/pull/174

### CODE REVIEW

File: src/api/stats/fantom/getHectorApy.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { addressBook } from '../../../../packages/address-book/address-book';
+const {
+  fantom: {
+    tokens: { WFTM },
+  },
+} = addressBook;
+
+const singlePool = [
+  {
+    name: 'curve-ftm-tor',
```
Comment (roman-monk) [inline src/api/stats/fantom/getHectorApy.ts:15]: please rename to hector-something as its not curve farm

File: src/api/stats/fantom/getHectorApy.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    address: '0x24699312CB27C26Cfc669459D670559E5E44EE60',
+    rewardPool: '0x61B71689684800f73eBb67378fc2e1527fbDC3b3',
+    chainId,
+  },
+];
+
+const getHectorApy = async () => {
```
Comment (roman-monk) [inline src/api/stats/fantom/getHectorApy.ts:22]: matter of taste but import of the address book just to set 1e18 for WFTM oracle seems too much code.

it's `1e18` by default, can just drop it:

```suggestion
const getHectorApy = async () =>
  getRewardPoolApys({
    pools: singlePool,
    oracleId: 'WFTM',
    oracle: 'tokens',
    web3: web3,
    chainId,
    // log: true,
  });
```

**Review by roman-monk — CHANGES_REQUESTED** [review]

File: src/api/stats/fantom/getHectorApy.ts (hunk 1 above)
Comment (0xCmdrKeen) [reply src/api/stats/fantom/getHectorApy.ts:15]: Does this apply to the entry in `src/data/fantom/curvePools.json` as well? I tried removing it, but that seems to break things. 

Also, for instance, `curve-ftm-mim` is named this way too even though it is staked on abracadabra.money, not Curve.

File: src/api/stats/fantom/getHectorApy.ts (hunk 2 above)
Comment (0xCmdrKeen) [reply src/api/stats/fantom/getHectorApy.ts:22]: This worked, but only when explicitly setting `decimals: 1e18`. Turns out there is no fallback for `params.decimals`, only `pool.decimals` (see https://github.com/beefyfinance/beefy-api/blob/52c4379f91892901ee77d1eb24d86a0699b3f5be/src/api/stats/common/getRewardPoolApys.js#L52)

File: src/api/stats/fantom/getHectorApy.ts (hunk 1 above)
Comment (roman-monk) [reply src/api/stats/fantom/getHectorApy.ts:7]: > Also, for instance, `curve-ftm-mim` is named this way too even though it is staked on abracadabra.money, not Curve.

no it's not

**Review by roman-monk — APPROVED** [review]

**Pull request merged**